### PR TITLE
[core-6650] Installing from sources - update based on latest changes

### PR DESCRIPTION
**Pull request by tinachoB**
# Installing from sources section update

## Description

The aim of these changes are to complement and update the new syscollector model. This one mainly involves changes in the building process adding new libraries to be taking into account in our documentation, primarily, on the `Installing from sources` section. 
A brief description of the epic: This epic aims to the integration of RSync and DBSync in syscollector, this is to make efficient use of the states that are captured today, in turn, it is important to clarify that these changes target the synchronization and do not alter the providers.
**Epic:** https://github.com/wazuh/wazuh/issues/6104
**Epic issue for documentation update:** https://github.com/wazuh/wazuh/issues/6650

To fulfill this mission, the refactoring of syscollector is not an option, but an obligation due to the characteristics and current design, in which there are a lot of coupling regarding the functionality (data collection, package creation, message sending)

## Checks
The following validations have been done on each Operating System updated in the documentation:
- **Manager**
  - [X] OpenSUSE: https://github.com/wazuh/wazuh/issues/6650#issuecomment-734993348
  - [X] Ubuntu: https://github.com/wazuh/wazuh/issues/6650#issuecomment-735006037
  - [X] CentOS: https://github.com/wazuh/wazuh/issues/6650#issuecomment-735810855

- **Agent**
  - [X] Ubuntu: https://github.com/wazuh/wazuh/issues/6650#issuecomment-735006037
  - [X] CentOS: https://github.com/wazuh/wazuh/issues/6650#issuecomment-735810855
  - [X] macOS: https://github.com/wazuh/wazuh/issues/6650#issuecomment-736015428
  - [X] Windows: https://github.com/wazuh/wazuh/issues/6650#issuecomment-736016068
  - [X] Solaris: https://github.com/wazuh/wazuh/issues/6650#issuecomment-737560783

## Note to the reviewer

This PR _should be merged_ once the [Epic 6104](https://github.com/wazuh/wazuh/issues/6104) is completed, tested, validated and merged into the corresponding Wazuh version (**v4.2.0**)

## Documentation update

**Manager**
image
image

**Agent**
image
image
image
image

### CODE REVIEW

**Comment by Dwordcito** [comment]
Hi @davidfmiranda Could you prioritize this issue?

File: source/user-manual/capabilities/syscollector.rst [hunk 1]
```diff
@@ -158,6 +162,8 @@ The current packages inventory of each Wazuh agent. On Linux systems, retrieved
 +------------------+----------------------------------------+---------------------------------------------------+-------------------+
 | **location**     | Location of the package                | C:\\Program Files\\VMware\\VMware Tools\\         | win/pkg           |
 +------------------+----------------------------------------+---------------------------------------------------+-------------------+
+| **checksum**     | Integrity synchronization value        | 78503709147600c8e0023cf2b9995772280eee30          | All               |
```
Comment (konangc87) [inline source/user-manual/capabilities/syscollector.rst:165]: missing item_id?

File: source/user-manual/capabilities/syscollector.rst [hunk 2]
```diff
@@ -171,43 +177,46 @@ it is composed of three tables to ensure that the information is as structured a
...
++------------------+---------------------------------+-----------------------------------------------------+-------------------+
+| **rx_errors**    | Reception errors                | 0                                                   | All               |
++------------------+---------------------------------+-----------------------------------------------------+-------------------+
+| **tx_dropped**   | Dropped transmission packets    | 0                                                   | All               |
++------------------+---------------------------------+-----------------------------------------------------+-------------------+
+| **rx_dropped**   | Dropped reception packets       | 0                                                   | All               |
++------------------+---------------------------------+-----------------------------------------------------+-------------------+
+| **checksum**     | Integrity synchronization value | 8503709147600c8e0023cf2b9995772280eee30             | All               |
++------------------+---------------------------------+-----------------------------------------------------+-------------------+
+
```
Comment (konangc87) [inline source/user-manual/capabilities/syscollector.rst:219]: ditto

File: source/user-manual/capabilities/syscollector.rst [hunk 3]
```diff
@@ -230,6 +239,8 @@ Referencing interfaces described at `sys_netiface`, this table shows the IPv4 an
 +------------------+---------------------------------+-----------------------------------------------------+-------------------+
 | **broadcast**    | Broadcast address               | 192.168.1.255                                       | All               |
 +------------------+---------------------------------+-----------------------------------------------------+-------------------+
+| **checksum**     | Integrity synchronization value | 78503709147600c8e0023cf2b9995772280eee30            | All               |
```
Comment (konangc87) [inline source/user-manual/capabilities/syscollector.rst:242]: ditto

File: source/user-manual/capabilities/syscollector.rst [hunk 4]
```diff
@@ -252,6 +263,8 @@ Referencing interfaces described at `sys_netiface`, this table shows the routing
 +------------------+---------------------------------+-----------------------------------------------------+---------------------+
 | **dhcp**         | DHCP status                     | enabled                                             | Linux/Windows       |
 +------------------+---------------------------------+-----------------------------------------------------+---------------------+
+| **checksum**     | Integrity synchronization value | 78503709147600c8e0023cf2b9995772280eee30            | All                 |
```
Comment (konangc87) [inline source/user-manual/capabilities/syscollector.rst:266]: ditto

File: source/user-manual/capabilities/syscollector.rst [hunk 5]
```diff
@@ -291,6 +304,8 @@ List the opened ports of a system.
 +------------------+----------------------------------------+---------------------------------------------------+-------------------+
 | **process**      | Name of the PID                        | System                                            | Windows/macOS     |
 +------------------+----------------------------------------+---------------------------------------------------+-------------------+
+| **checksum**     | Integrity synchronization value        | 78503709147600c8e0023cf2b9995772280eee30          | All               |
```
Comment (konangc87) [inline source/user-manual/capabilities/syscollector.rst:307]: ditto

**Review by konangc87 — CHANGES_REQUESTED** [review]

File: source/installation-guide/more-installation-alternatives/wazuh-from-sources/wazuh-agent/index.rst [hunk 6]
```diff
@@ -14,31 +14,78 @@ Installing Wazuh agent from sources
...
+              # yum install make gcc gcc-c++ python3 python3-policycoreutils automake autoconf libtool
+              # rpm -i http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/libstdc++-static-8.3.1-5.1.el8.x86_64.rpm
+
+            CMake 3.18 installation
 
+            .. code-block:: console
+
+              # curl -OL http://packages.wazuh.com/utils/cmake/cmake-3.18.3.tar.gz && tar -zxf cmake-3.18.3.tar.gz
+              # cd cmake-3.18.3 && ./bootstrap --no-system-curl
+              # make -j$(nproc) && make install
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply source/installation-guide/more-installation-alternatives/wazuh-from-sources/wazuh-agent/index.rst:56]: Will do!

File: source/installation-guide/more-installation-alternatives/wazuh-from-sources/wazuh-agent/index.rst [hunk 7]
```diff
@@ -553,47 +599,120 @@ Installing Wazuh agent from sources
...
+            # gmake
+            # gmake install
+            # echo "export PATH=/usr/local/gcc-5.5.0/bin:${PATH}" >> /etc/profile
+            # PATH="/usr/local/gcc-5.5.0/bin:${PATH}"
+            # export PATH
+            # CPLUS_INCLUDE_PATH=/usr/local/gcc-5.5.0/include/c++/5.5.0/
+            # export CPLUS_INCLUDE_PATH
+            # LD_LIBRARY_PATH=/usr/local/gcc-5.5.0/lib/
+            # export LD_LIBRARY_PATH
+            # cd ..
```
[the remark this answers is not in the document]
Comment (tinachoB) [reply source/installation-guide/more-installation-alternatives/wazuh-from-sources/wazuh-agent/index.rst:645]: Will do!

File: source/user-manual/capabilities/syscollector.rst (hunk 1 above)
Comment (tinachoB) [reply source/user-manual/capabilities/syscollector.rst:165]: Yup. Will add it.

File: source/user-manual/capabilities/syscollector.rst (hunk 2 above)
Comment (tinachoB) [reply source/user-manual/capabilities/syscollector.rst:219]: Yup. Will add it.

File: source/user-manual/capabilities/syscollector.rst (hunk 4 above)
Comment (tinachoB) [reply source/user-manual/capabilities/syscollector.rst:266]: Yup. Will add it.

File: source/user-manual/capabilities/syscollector.rst (hunk 5 above)
Comment (tinachoB) [reply source/user-manual/capabilities/syscollector.rst:307]: Yup. Will add it.

**Review by konangc87 — APPROVED** [review]

**Review by davidfmiranda — APPROVED** [review]
LGTM

**Comment by tinachoB** [comment]
@vikman90 - if you are ok with these changes, please go ahead and merge it! Thank you!

**Review by vikman90 — APPROVED** [review]
Hi team,

Can we find an alter

**Pull request merged**